src/pages/debrief/components/vehicle-checks-card/vehicle-checks-card.ts

```diff
@@ -17,4 +17,12 @@ export class VehicleChecksCardComponent {
   constructor() { }
 
   questionHasFault = (result: QuestionResult): boolean => result.outcome !== CompetencyOutcome.P;
+
+  ngOnInit(): void {
+    this.tellMeShowMeQuestions.forEach((value: QuestionResult, index: number, array: QuestionResult[]): void => {
+      if (!value.hasOwnProperty('outcome')) {
+        this.tellMeShowMeQuestions.splice(index, 1);
+      }
```

Comment:
Splicing an array as your iterating through it feels a bit risky to me. A .filter would be a lot safer to remove the elements instead of using a .foreach

Reply:
You are completely right, it'll skip an index after the splice.. I don't know how I missed that, nice spot, will be amended soon.